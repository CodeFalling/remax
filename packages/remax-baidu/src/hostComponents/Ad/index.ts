import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AdProps extends BaseProps {
  appid?: string;
  apid?: string;
  type?: 'feed' | 'banner';
  updatetime?: string;
  onError?: (event: GenericEvent) => void;
  onLoad?: (event: GenericEvent) => void;
  onClose?: (event: GenericEvent) => void;
  onStatus?: (event: GenericEvent) => void;
}

export const Ad: React.ComponentType<AdProps> = createHostComponent<AdProps>('ad');

Ad.defaultProps = {
  type: 'feed',
};
