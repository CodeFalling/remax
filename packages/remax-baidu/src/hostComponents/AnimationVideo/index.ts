import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AnimationVideoProps extends BaseProps {
  resourceWidth?: number;
  resourceHeight?: number;
  canvasStyle?: React.CSSProperties;
  path?: string;
  loop?: boolean;
  autoPlay?: boolean;
  onStarted?: (event: GenericEvent) => void;
  onEnded?: (event: GenericEvent) => void;
}

export const AnimationVideo: React.ComponentType<AnimationVideoProps> = createHostComponent<AnimationVideoProps>(
  'animation-video'
);

AnimationVideo.defaultProps = {
  resourceWidth: 800,
  resourceHeight: 400,
  canvasStyle: {
    width: '400px',
    height: '400px',
  },
  loop: false,
  autoPlay: false,
};
