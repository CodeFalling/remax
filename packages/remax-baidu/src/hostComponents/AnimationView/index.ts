import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AnimationViewProps extends BaseProps {
  path?: string;
  loop?: boolean;
  autoPlay?: boolean;
  action?: 'play' | 'pause' | 'stop';
  hidden?: boolean;
  onEnded?: (event: GenericEvent) => void;
}

export const AnimationView: React.ComponentType<AnimationViewProps> = createHostComponent<AnimationViewProps>(
  'animation-view'
);

AnimationView.defaultProps = {
  loop: false,
  autoPlay: true,
  action: 'play',
  hidden: true,
};
