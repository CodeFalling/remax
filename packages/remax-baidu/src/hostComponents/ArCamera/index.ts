import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface ArCameraProps extends BaseProps {
  key?: string;
  type?: 0 | 5 | 8;
  flash?: 'auto' | 'on' | 'off';
  onError?: (event: GenericEvent) => void;
  onLoad?: (event: GenericEvent) => void;
  onMessage?: (event: GenericEvent) => void;
  onScancode?: (event: GenericEvent) => void;
}

export const ArCamera: React.ComponentType<ArCameraProps> = createHostComponent<ArCameraProps>('ar-camera');

ArCamera.defaultProps = {
  flash: 'auto',
  type: 0,
};
