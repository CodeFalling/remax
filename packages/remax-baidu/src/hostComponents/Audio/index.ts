import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface AudioProps extends BaseProps {
  src?: string;
  loop?: boolean;
  controls?: boolean;
  poster?: string;
  name?: string;
  author?: string;
  onError?: (event: GenericEvent) => void;
  onPlay?: (event: GenericEvent) => void;
  onPause?: (event: GenericEvent) => void;
  onTimeUpdate?: (event: GenericEvent) => void;
  onEnded?: (event: GenericEvent) => void;
}

export const Audio: React.ComponentType<AudioProps> = createHostComponent<AudioProps>('audio');

Audio.defaultProps = {
  loop: false,
  controls: false,
};
