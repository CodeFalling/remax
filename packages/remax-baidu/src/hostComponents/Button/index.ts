import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps } from '../../types/component';

export interface ButtonProps extends BaseProps {
  size?: 'default' | 'mini';
  type?: 'primary' | 'default' | 'warn';
  plain?: boolean;
  formType?: 'submit' | 'reset' | 'buttonclick';
  openType?:
    | 'share'
    | 'getUserInfo'
    | 'getPhoneNumber'
    | 'openSetting'
    | 'chooseAddress'
    | 'chooseInvoiceTitle'
    | 'contact';
  hoverClassName?: string;
  hoverStopPropagation?: boolean;
  hoverStartTime?: number;
  hoverStayTime?: number;
  disabled?: boolean;
  loading?: boolean;
  onGetPhoneNumber?: (event: any) => void;
  onGetUserInfo?: (event: any) => void;
  onOpenSetting?: (event: any) => void;
  onContact?: (event: any) => void;
  onChooseAddress?: (event: any) => void;
  onChooseInvoiceTitle?: (event: any) => void;
  contact?: (event: any) => void;
}

export const Button: React.ComponentType<ButtonProps> = createHostComponent<ButtonProps>('button');

Button.defaultProps = {
  hoverClassName: 'button-hover',
  hoverStartTime: 20,
  hoverStayTime: 70,
  hoverStopPropagation: false,
  disabled: false,
  loading: false,
  size: 'default',
  type: 'default',
  plain: false,
  formType: 'buttonclick',
};
