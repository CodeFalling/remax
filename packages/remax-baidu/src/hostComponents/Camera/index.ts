import * as React from 'react';
import { createHostComponent } from '@remax/shared';
import { BaseProps, GenericEvent } from '../../types/component';

export interface CameraProps extends BaseProps {
  devicePosition?: 'back' | 'front';
  flash?: 'auto' | 'on' | 'off';
  onStop?: (event: GenericEvent) => void;
  onError?: (event: GenericEvent) => void;
}

export const Camera: React.ComponentType<CameraProps> = createHostComponent<CameraProps>('camera');

Camera.defaultProps = {
  devicePosition: 'back',
  flash: 'auto',
};
