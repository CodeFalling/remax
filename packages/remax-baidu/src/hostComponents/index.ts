export * from './Ad';
export * from './AnimationVideo';
export * from './AnimationView';
export * from './Audio';
export * from './ArCamera';
export * from './Camera';
export * from './Button';
export * from './Canvas';
export * from './Checkbox';
export * from './CheckboxGroup';
export * from './CoverImage';
export * from './CoverView';
export * from './Form';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Label';
export * from './LivePlayer';
export * from './MovableArea';
export * from './MovableView';
export * from './Navigator';
export * from './OpenData';
export * from './Picker';
export * from './PickerView';
export * from './PickerViewColumn';
export * from './Progress';
export * from './Radio';
export * from './RadioGroup';
export * from './RichText';
export * from './RtcRoom';
export * from './RtcRoomItem';
export * from './ScrollView';
export * from './Slider';
export * from './Swiper';
export * from './SwiperItem';
export * from './Map';
export * from './Switch';
export * from './Tabs';
export * from './TabItem';
export * from './Text';
export * from './Textarea';
export * from './Video';
export * from './View';
export * from './WebView';
