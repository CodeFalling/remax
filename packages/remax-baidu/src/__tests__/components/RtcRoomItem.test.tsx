import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { RtcRoomItem } from '../../hostComponents';

describe('RtcRoomItem', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<RtcRoomItem />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
