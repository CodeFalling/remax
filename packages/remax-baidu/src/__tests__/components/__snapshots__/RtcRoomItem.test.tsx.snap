// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RtcRoomItem render correctly 1`] = `<rtc-room-item />`;
